pallets/xyk/src/lib.rs

```diff
@@ -1614,8 +1615,10 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 				sold_asset_id.into(),
 				&sender,
 				&vault,
-				(sold_asset_amount - buy_and_burn_amount - treasury_amount - pool_fee_amount)
-					.into(),
+				(sold_asset_amount
+					.checked_sub(buy_and_burn_amount + treasury_amount + pool_fee_amount)
```

Comment:
what about possible overflow with `+`. Shouldt we use checked_add everywhere ?

Reply:
buy_and_burn_amount and the rest are 0.05% of sold amount which is capped u128, so no need